lib/store_spec.ml

```diff
@@ -24,39 +24,59 @@ let pp f = function
 
 type store = Store : (module S.STORE with type t = 'a) * 'a -> store
 
-let to_store mode = function
+let to_store rsync_mode = function
   | `Btrfs path ->
     Btrfs_store.create path >|= fun store ->
     Store ((module Btrfs_store), store)
   | `Zfs pool ->
     Zfs_store.create ~pool >|= fun store ->
     Store ((module Zfs_store), store)
   | `Rsync path ->
-    Rsync_store.create ~path ~mode () >|= fun store ->
+    Rsync_store.create ~path ~mode:rsync_mode () >|= fun store ->
     Store ((module Rsync_store), store)
 
-let cmdliner =
-  let open Cmdliner in
-  let store_t = Arg.conv (of_string, pp) in
-  let store =
-    Arg.required @@
-    Arg.opt Arg.(some store_t) None @@
-    Arg.info
-      ~doc:"$(b,btrfs:/path) or $(b,rsync:/path) or $(b,zfs:pool) for build cache."
-      ~docv:"STORE"
-      ["store"]
-  in
-  let rsync_mode =
-    let options =
-      [("copy", Rsync_store.Copy);
-       ("hardlink", Rsync_store.Hardlink);
-       ("hardlink_unsafe", Rsync_store.Hardlink_unsafe)]
-    in
-    Arg.value @@
-    Arg.opt (Arg.enum options) Rsync_store.Copy @@
-    Arg.info
-      ~doc:"$(b,copy) or $(b,hardlink), to optimize for speed or low disk usage."
-      ~docv:"RSYNC_MODE"
-      ["rsync-mode"]
+open Cmdliner
+
+let store_t = Arg.conv (of_string, pp)
+
+let store =
+  Arg.required @@
+  Arg.opt Arg.(some store_t) None @@
+  Arg.info
+    ~doc:"$(docv) must be one of $(b,btrfs:/path), $(b,rsync:/path) or $(b,zfs:pool) for the build cache."
+    ~docv:"STORE"
+    ["store"]
+
+let store' =
+  Arg.value @@
+  Arg.opt Arg.(some store_t) None @@
+  Arg.info
+    ~doc:"$(docv) must be one of $(b,btrfs:/path), $(b,rsync:/path) or $(b,zfs:pool) for the OBuilder cache."
+    ~docv:"STORE"
+    ["obuilder-store"]
```

Comment:
There doesn't seem to be a reason to have 2 copies of this code.

```suggestion
let store =
  Arg.opt Arg.(some store_t) None @@
  Arg.info
    ~doc:"$(docv) must be one of $(b,btrfs:/path), $(b,rsync:/path) or $(b,zfs:pool) for the OBuilder cache."
    ~docv:"STORE"
    ["store"]
```

then combining it with `Arg.value` or `Arg.required` at the call site.

Reply:
> There doesn't seem to be a reason to have 2 copies of this code.

I also want to keep the option `--obuilder-store` for current compatibility with ocluster-worker. Maybe the API should expose the names of the option?